### PR TITLE
feat(sdk): capsule install/upgrade lifecycle hooks and elicit API

### PR DESCRIPTION
**Pull request by joshuajbouw**
## Summary

- Add `#[astrid::install]` and `#[astrid::upgrade]` proc macro attributes to `astrid-sdk-macros` with compile-time validation: signature enforcement (`&self` for install, `&self, &str` for upgrade), duplicate detection, `#[astrid::mutable]` rejection on lifecycle hooks, and `&str` type checking via `let`-chain inspection
- Add `elicit` module to `astrid-sdk` with `secret()`, `has_secret()`, `text()`, `text_with_default()`, `select()`, and `array()` functions for user input during lifecycle hooks. `secret()` never returns the value to the capsule. Input validation includes empty/whitespace key rejection, empty options rejection on `select()`, and defense-in-depth response validation
- Add `astrid_elicit` and `astrid_has_secret` FFI declarations to `astrid-sys`. Conditional WASM export generation: capsules without lifecycle attributes are completely unaffected. Stateful capsules correctly persist/load state through KV

## Test Plan

- `cargo test -p astrid-sdk-macros` - 20 tests covering all compile-time validation paths, stateful vs non-stateful persistence codegen, attribute interaction edge cases, and export generation
- `cargo clippy --workspace -- -D warnings` clean
- `cargo fmt --check` clean

## Future Work (out of scope)

- Kernel-side host function registration for `astrid_elicit` / `astrid_has_secret` (issue #312)
- Kernel-side `astrid_install` / `astrid_upgrade` dispatch during capsule load
- `SecretStore` implementation (OS keychain + encrypted file fallback) - separate PR in `astrid-crypto`
- HTTP secret injection via `[secrets]` manifest config

### CODE REVIEW

File: crates/astrid-sdk-macros/src/lib.rs [hunk 1]
```diff
@@ -206,6 +271,113 @@ fn capsule_impl(
...
+                        let err = format!("{:?}", rc.0);
+                        if let Ok(mut mem) = ::extism_pdk::Memory::from_bytes(&err) {
+                            unsafe { ::extism_pdk::extism::error_set(mem.offset()); }
+                        }
+                        return rc.1;
+                    }
+                };
+                ::extism_pdk::unwrap!(::extism_pdk::output(&output));
+                0
+            }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-sdk-macros/src/lib.rs:322]: @gemini-code-assist Acknowledged, but intentionally not changing here. This boilerplate pattern is pre-existing across all 5 existing exports (astrid_tool_call, astrid_command_run, astrid_hook_trigger, astrid_cron_trigger, astrid_export_schemas). Extracting a helper for just the 2 new ones would create inconsistency. Worth doing as a separate refactor PR across all 7 exports.

File: crates/astrid-sdk-macros/src/lib.rs [hunk 2]
```diff
@@ -206,6 +271,113 @@ fn capsule_impl(
...
+    let upgrade_export = upgrade_method.map(|method_name| {
+        let body = if is_stateful {
+            quote! {
+                // JsonError covers key-not-found (host returns empty bytes which
+                // fail to parse) and corrupt state - both fall back to Default.
+                let mut instance: #struct_name = match ::astrid_sdk::prelude::kv::get_json("__state") {
+                    Ok(state) => state,
+                    Err(::astrid_sdk::SysError::JsonError(_)) => Default::default(),
+                    Err(e) => return Err(::extism_pdk::Error::msg(format!("failed to load state: {}", e))),
+                };
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-sdk-macros/src/lib.rs:342]: @gemini-code-assist Good catch. Fixed in 832a7dc - the JsonError fallback now logs a warning via `sys::log("warn", ...)` before falling back to Default. Silent state corruption was a real debugging trap.

File: crates/astrid-sdk/src/lib.rs [hunk 3]
```diff
@@ -351,8 +351,180 @@ pub mod process {
...
+        validate_key(key)?;
+        let req = ElicitRequest {
+            kind: "secret",
+            key,
+            description: Some(description),
+            options: None,
+            default: None,
+        };
+        let req_bytes = serde_json::to_vec(&req)?;
+        let resp_bytes = unsafe { astrid_elicit(req_bytes)? };
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-sdk/src/lib.rs:398]: @gemini-code-assist Intentionally not adding here. The existing codebase has ~20 unsafe FFI calls in this same file (fs::exists, fs::read_bytes, kv::get_bytes, etc.) and none have SAFETY comments - they all follow the same pattern of calling Extism host functions. Adding them only to the new elicit calls would be inconsistent. If we add SAFETY comments, it should be a codebase-wide pass in a separate PR.

File: crates/astrid-sdk/src/lib.rs [hunk 4]
```diff
@@ -351,8 +351,180 @@ pub mod process {
...
+                "kernel did not confirm secret storage".into(),
+            ));
+        }
+        Ok(())
+    }
+
+    /// Check if a secret has been configured (without reading it).
+    pub fn has_secret(key: &str) -> Result<bool, SysError> {
+        validate_key(key)?;
+        let req_bytes = serde_json::to_vec(&serde_json::json!({"key": key}))?;
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-sdk/src/lib.rs]: @gemini-code-assist Good point. Fixed in 832a7dc - now uses a typed `HasSecretRequest` struct instead of `json!` macro, avoiding the intermediate `serde_json::Value` allocation.

File: crates/astrid-sdk/src/lib.rs [hunk 5]
```diff
@@ -351,8 +351,180 @@ pub mod process {
...
+        let req_bytes = serde_json::to_vec(&req)?;
+        let resp_bytes = unsafe { astrid_elicit(req_bytes)? };
+
+        #[derive(serde::Deserialize)]
+        struct TextResp {
+            value: String,
+        }
+        let resp: TextResp = serde_json::from_slice(&resp_bytes)?;
+        Ok(resp.value)
+    }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-sdk/src/lib.rs]: @gemini-code-assist Agreed. Fixed in 832a7dc - extracted a private `elicit_text(key, description, default: Option<&str>)` helper. `text()` and `text_with_default()` are now thin wrappers passing `None` / `Some(default)`.

File: crates/astrid-sdk/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-sdk/src/lib.rs:398]: @gemini-code-assist Added in d9ef825. SAFETY comments on all 5 unsafe FFI calls in the elicit module.
